tests/ui/consts/const-eval/heap/alloc_intrinsic_untyped.rs

```diff
@@ -1,11 +1,12 @@
+// We unleash Miri here since this test demonstrates code that bypasses the checks against interning
+// mutable pointers, which currently ICEs. Unleashing Miri silence the ICE.
```

Comment:
I'd like to see a test in `tests/crashes` to track this (to make sure that it does ICE with the current state of this PR)